include/openssl/bio.h

```diff
@@ -156,11 +156,9 @@ extern "C" {
 
 # ifndef OPENSSL_NO_KTLS
 #  define BIO_get_ktls_send(b)         \
-     (BIO_method_type(b) == BIO_TYPE_SOCKET \
-      && BIO_ctrl(b, BIO_CTRL_GET_KTLS_SEND, 0, NULL))
+     BIO_ctrl(b, BIO_CTRL_GET_KTLS_SEND, 0, NULL)
 #  define BIO_get_ktls_recv(b)         \
-     (BIO_method_type(b) == BIO_TYPE_SOCKET \
-      && BIO_ctrl(b, BIO_CTRL_GET_KTLS_RECV, 0, NULL))
+     BIO_ctrl(b, BIO_CTRL_GET_KTLS_RECV, 0, NULL)
```

Comment:
I recall the change to add the BIO_TYPE_SOCKET check was made due to a problem in the BoringSSL tests. On reflection it was probably the wrong change - but I suspect we may see BoringSSL test failures again as a result of this.

Reply:
I spent some time trying to get the BoringSSL tests to run but I haven't succeeded in getting them to run to try to investigate this further.  The current tag for the boringssl submodule points to an older version of boringssl whose test runner does not work with the current version of Go (1.13) seemingly due to the transition to modules instead of GOPATH in 1.13?  (I'm not very familiar with Go, but 'go get' in the ssl/test/runner subdirectory refused to be happy despite my efforts).

I then tried to update to the master of BoringSSL to get a version that works with modern Go.  After fixing the BSD configuration of OpenSSL to enable C++ so that ossl_shim is built, I found that the ossl_shim in the tree doesn't support some option that boringssl master wants (-is-handshaker-supported).  Also, 'make test' doesn't set LD_LIBRARY_PATH when invoking go so ossl_shim doesn't run from 'make test' since it doesn't find the libssl.so.3 that was built (and setting LD_LIBRARY_PATH in the environment passed to make doesn't fix this, perhaps go is stripping it from the environment before invoking the shim).  However, running the shim directly I can see it doesn't support the option.

I can submit another PR to enable C++ on BSD targets either in the PR or a separate PR, but I'm at a bit of a loss to get the boringssl tests to actually run.

Reply:
Yes, I suspect there may be other boringssl issues too. Probably I need to spend some time looking at this - but its not made it to the top of the TODO list. I don't think you need to worry about it WRT to this change.